packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,215 @@
+import { Address, Customer, PaymentSessionStatus } from "../models"
+import { MedusaContainer } from "../types/global"
+
+export type PaymentProcessorContext = {
+  billing_address?: Address | null
+  email: string
+  currency_code: string
+  amount: number
+  resource_id?: string
+  customer?: Customer
+  context: Record<string, unknown>
+}
+
+export type PaymentProcessorSessionResponse = {
+  update_requests: { customer_metadata: Record<string, unknown> }
+  session_data: Record<string, unknown>
+}
+
+export interface PaymentProcessorError {
+  error: string
+  code: number
+  details: any
+}
+
+/**
+ * The new payment service plugin interface
+ * This work is still experimental and can be changed until it becomes stable
+ */
+export interface PaymentProcessor {
+  /**
+   * Return a unique identifier to retrieve the payment plugin provider
+   */
+  getIdentifier(): string
+
+  /**
+   * Used to initialise anything like an SDK or similar
+   */
+  init(): Promise<void>
+
+  /**
+   * Initiate a payment session with the external provider
+   */
+  initiatePayment(
+    context: PaymentProcessorContext
+  ): Promise<PaymentProcessorError | PaymentProcessorSessionResponse>
+
+  /**
+   * Update an existing payment session
+   * @param paymentId
+   * @param context
+   */
+  updatePayment(
+    paymentId: string,
```

Comment:
@olivermrbl @srindom I was working on the next step (loader and payment provider support) and I came to think that we have no idea of what is the paymentId specifically. In fact, we are saving the data in the payment session data and only the plugin is aware of where is the payment id located and what is the name of the property. For stripe it is the `id` from the session data, but it seems wrong to assume that it is always the case. What do you think if instead, the first argument is the payment session data?